orttraining/orttraining/training_ops/cuda/communication/nccl_service.cu

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#if defined(USE_NCCL) || defined(USE_HOROVOD)
+
+#include "orttraining/training_ops/cuda/communication/nccl_service.cuh"
+#include "core/profile/context.h"
+#include <iostream>
+#include <nccl.h>
+
+namespace onnxruntime {
+namespace cuda {
+
+void NcclService::SubmitSendAndWait(void* ptr, size_t size, int peer) {
```

Comment:
>NcclService [](start = 5, length = 11)

This file seems should be named nccl_service.cc, as it does not contain any CUDA kernels with __device__.

Reply:
Ok.

---
In reply to: [483802659](https://github.com/microsoft/onnxruntime/pull/5054#discussion_r483802659) [](ancestors = 483802659)